guides/micronaut-oracle-cloud/micronaut-oracle-cloud.adoc

```diff
@@ -0,0 +1,171 @@
+
+== What you will need
+
+To complete this guide, you will need the following:
+
+* Some time on your hands
+* A decent text editor or IDE
+* JDK 1.8 or greater installed with `JAVA_HOME` configured appropriately
+* An Oracle Cloud account (create a free trial account at https://signup.oraclecloud.com[signup.oraclecloud.com])
+
+include::{commondir}/common-completesolution.adoc[]
+
+include::{commondir}/common-create-app.adoc[]
+
+== Sample Controller
+
+Next we'll create a simple controller that we can use to access the application once it's deployed.
+
+source:HelloController[]
+
+== Create an Oracle Cloud Compute Instance
+
+Login to your Oracle Cloud tenancy, and from the Oracle Cloud Home Page select "Create a VM Instance": +
+
+image::oraclecloud.createvm.1.png[]
+
+Enter "micronaut-demo" as the name of the instance leaving the "Placement" and "Shape" as "Always Free Eligible" (or any small image shape): +
+
+image::oraclecloud.createvm.2.png[]
+
+Under "Add SSH keys" choose "Generate SSH key pair" and then click "Save Private Key" to save the private key locally on disk:
+
+image::oraclecloud.createvm.3.png[]
+
+The private key will download as a file named "ssh-key-(date).key".
+
+Leave the remaining settings as their defaults and click the "Create" button:
+
+image::oraclecloud.createvm.4.png[]
+
+Take note of the IP address of your instance - click the "Copy" link next to "Public IP Address".
+
+image::oraclecloud.createvm.5.png[]
+
+Click on the "Subnet" link ("Default Subnet ..."):
+
+image::oraclecloud.createvm.6.png[]
+
+Under "Security Lists" click on the "Default Security List":
+
+image::oraclecloud.createvm.7.png[]
+
+Click "Add Ingress Rules":
+
+image::oraclecloud.createvm.8.png[]
+
+Enter "0.0.0.0/0" as the "Source CIDR", 8080 as the "Destination Port Range", and "micronaut" as the Description, and click "Add Ingress Rules".
+
+image::oraclecloud.createvm.9.png[]
+
+== Deploy to Oracle Cloud
+
+First ensure that the private key you downloaded has the correct permissions:
+
+[source, bash]
+----
+$ chmod 400 /path/to/ssh-key-*.key
+----
+
+Create an executable jar including all dependencies:
+
+:exclude-for-build:maven
+
+[source,bash]
+----
+$ ./gradlew assemble
+----
+
+:exclude-for-build:
+
+:exclude-for-build:gradle
+
+[source,bash]
+----
+$ ./mvnw package
+----
+
+:exclude-for-build:
+
+Push the JAR file to your VM:
+
+:exclude-for-build:maven
+
+[source,bash]
+----
+$ scp -i /path/to/ssh-key-*.key build/libs/micronautguide-0.1-all.jar opc@[VM IP Address]:/home/opc/application.jar
+----
+
+:exclude-for-build:
+
+:exclude-for-build:gradle
+
+[source,bash]
+----
+$ scp -i /path/to/ssh-key-*.key target/micronautguide-0.1.jar opc@[VM IP Address]:/home/opc/application.jar
+----
+
+:exclude-for-build:
+
+To run on the VM, first SSH in:
+
+[source,bash]
+----
+$ ssh -i /path/to/ssh-key-*.key opc@[YOUR IP]
+----
+
+Then install Java:
+
+include::{commondir}/common-install-graalvm-ee.adoc[]
```

Comment:
For Groovy, there will be no instructions since the content of `common-install-grallvm-ee.adoc` does not contain anything for groovy.

Reply:
I think these exclusions should be removed https://github.com/micronaut-projects/micronaut-guides-poc/pull/198/files#diff-1fd880d0d3823f79435dc79f2ebab49661cfd376e079fa926487dff978698a72R1

There is no native image instructions in this guide

Reply:
> I think these exclusions should be removed [#198 (files)](https://github.com/micronaut-projects/micronaut-guides-poc/pull/198/files#diff-1fd880d0d3823f79435dc79f2ebab49661cfd376e079fa926487dff978698a72R1)
> 
> There is no native image instructions in this guide

The code is in a "common snippet" that can be used anywhere. It makes sense to "guard" that code for Groovy guides. 

Actually the guide is created for the 3 languages so it should be changed to:
a) Only use Java/Kotlin
b) "Guard" the part about the native image when using Groovy

Reply:
Ummm, sorry about this. Graeme is right (and I need to learn how to read proper English). The guide has nothing to do with GraalVM, so yes. That exclusions for Groovy (that I added this morning) need to be deleted.
Sorry for the noise @burtbeckwith